packages/crud/src/crud/CrudTableHeader.tsx

```diff
@@ -107,15 +108,27 @@ const CrudTableHeader: React.FC<CrudTableHeaderProps> = (props) => {
     const nextFilters = Object.entries(appliedFilters).reduce(
       (acc, [key, value]) => {
         const fieldDef = get(filterAndSearchFormFieldDefs, key)
-
         // allow custom formatting
         if (fieldDef?.setFilterQuery) {
           const [formattedKey, formattedValue] = fieldDef.setFilterQuery([
             key,
-            value,
+            value
           ])
 
-          return { ...acc, [formattedKey]: formattedValue }
+          const nextFilters = {
+            ...acc,
+            [formattedKey]: formattedValue,
+            ...(typeof value !== 'object' && { [key]: value })
+          }
+
+          if (key.endsWith('_id')) {
+            const relationalObjectKey = getRelationalObjectKey(key)
+            assign(nextFilters, {
+              [relationalObjectKey]: get(appliedFilters, relationalObjectKey)
+            })
+          }
```

Comment:
@clodal this is for custom formatting of the query key and value for SERVER-SIDE FILTERING, however, we still need to retain the original key and value pair for FilterForm to display correctly
E.g For `lines.order_form_line.order_form.sales_order.project_id=2`, we still need to add `project_id=2` and `project=Fusheng House`. `project_id` is for FilterForm, and `project` is for the filter chip